src/configuration.c

```diff
@@ -1197,7 +1198,9 @@ config_scan_pem_dir(char *pemdir, hitch_config *cfg)
 			if (fnmatch(cfg->PEM_DIR_GLOB, d[i]->d_name, 0))
 				continue;
 		}
-		if (d[i]->d_type != DT_REG)
+		if (d[i]->d_type != DT_UNKNOWN && d[i]->d_type != DT_REG)
+			/* The underlying filesystem supports d_type but the current file
+			 * is not a regular one: skip it right now. */
```

Comment:
It looks like the comment is slightly off, and in general I'm not in favor of comments paraphrasing code, I would simply remove the comment.

Reply:
I can understand your feeling about comments paraphrasing code. However, before I adjust this part, I would like to better understand: how is this comment "slightly off"?

Reply:
Because the comment describes what happened before your change, skipping non-regular files. With your change we fall back to `lstat(2)` to figure that out in the unknown case. So from my point of view the comment paraphrases the previous behavior and is slightly off because it doesn't explain the unknown special case, and in that regard regular DT_UNKNOWN files are not skipped "right now".

A comment could have said something like:

> We only match regular files but if the filesystem doesn't provide that information we later perform this check with `lstat(2)`.

But looking at the code, I don't feel we need a comment to explain what is happening.

I hope that clarifies my pedantic review, no offense meant. By the way, good catch!

Reply:
> I hope that clarifies my pedantic review, no offense meant. 
Thanks for the clarification, no offense taken.

> But looking at the code, I don't feel we need a comment to explain what is happening.
Fine with me -- I have removed this comment.